WcaOnRails/app/webpacker/lib/requests/routes.js.erb

```diff
@@ -11,8 +11,27 @@ export const postsUrl = (page, format = 'json') => {
   return `<%= CGI.unescape(Rails.application.routes.url_helpers.posts_path(format: "${format}")) %>?page=${page}`;
 };
 
+export const incidentsUrl = (perPage, page, tags = undefined, searchString = undefined, competitions = undefined, format = 'json') => {
+  let baseUrl = `<%= CGI.unescape(Rails.application.routes.url_helpers.incidents_path(format: "${format}")) %>?per_page=${perPage}&page=${page}`;
+  if (tags && tags.length > 0) {
+    baseUrl += `&tags=${tags}`;
+  }
+  if (competitions) {
+    baseUrl += `&competitions=${competitions}`;
+  }
+  if (searchString) {
+    baseUrl += `&q=${searchString}`
+  }
+  return baseUrl;
```

Comment:
Can this be replaced with a usage of URLSearchparams.
You should be able to initiate a new object of it setting the tags, competitions, and q.

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams